evalml/tests/pipeline_tests/test_pipelines.py

```diff
@@ -2754,3 +2755,80 @@ def test_pipeline_transform_with_final_estimator(
         ),
     ):
         pipeline.transform(X, y)
+
+
+@patch("evalml.pipelines.components.LogisticRegressionClassifier.fit")
```

Comment:
@angela97lin Can you please add coverage for `fit_features` and `compute_estimator_features`?

Reply:
The expectation is that the training only components are run during `fit_features` and not `compute_estimator_features`

Reply:
Of course, great idea 😁